lr_listener.go

```diff
@@ -230,7 +230,7 @@ func (l *LogicalReplicationListener) processMessage(msg *pgx.ReplicationMessage)
 		l.errCh <- fmt.Errorf("failed to parse LSN from wal2json message: %v", err)
 		return
 	}
-	if lsn > l.replLSN {
+	if lsn >= l.replLSN {
```

Comment:
I think this ok. But just want to double check. Was it just the log message that was wrong? Or the logic? If LSN is current then there should be nothing to do right?

Reply:
Thinking out loud for this:
* `sourceDB` should send (batches? single?) WAL messages in-order
* we are running a single loop for `listener.processMessage()`, so there's no race condition for parallel processing
* if we receive an old LSN, we don't have enough information to 'correct' for it mid processing and investigation should happen

With that logic this should be an error that halts processing - coming up with a `fixup` now

Reply:
This one I am still not completely sure about. Would `=` mean it is up-to-date and there is nothing to do (no error). And `<` should not occur, so something is wrong (error)?

Reply:
My understanding is that `=` means something is being reprocessed - this `processMessage` codepath is only executed on messages that have WAL content, so any LSN that is not greater than what we have means that we have already processed it.

Reply:
Whether that results in `log a warning and keep listening` or `shut it down because something unpredictable is happening` is up for debate (I err on the side of `shut it down`)

Reply:
Ok. I agree that exiting with an error is the safest and best approach for now.